### ISSUE TITLE
more flexible color specification?

### ISSUE THREAD

**Issue by Phillip-M-Feldman** [ipython/ipython#12270]
I've been experimenting with values for `c.InteractiveShell.colors`.  The best option appears to be 'Linux', but even with that choice the contrast is still quite poor: operators like + and - appear in purple, and are almost unreadable against the black background.  It would be helpful if `c.InteractiveShell.Colors` allowed one to submit either a list or dictionary of html color specifiers, with one specifier for each type of element (background, Python reserved words, variable names, numbers, and operators).

**Comment by Carreau** [comment]
You actually can, It needs more info on the documentation (pull requests welcome)  but you can completely overwrite the style on a per token basis. https://github.com/ipython/ipython/issues/12188 have some info. You can use the pygments way of specifying colors, meaning that you can either do hexadecimal notation, or use the ansi naming to respect your terminal emulator theme. 

The defaults colors are really hard to get rights, and we try to avoid platform specific hack as they are unreliable.

You should even be able to set any [highlighting theme from pygments](https://help.farbox.com/pygments.html) (though they won't affect, prompt and traceback IIRC). 

<img width="384" alt="Screen Shot 2020-04-30 at 20 14 14" src="https://user-images.githubusercontent.com/335567/80779015-3359ed80-8b1f-11ea-8139-59e7bf9fe529.png">

**Comment by Carreau** [comment]
(also yes in general a better way of listing styles and affecting both code highlighting and prompt traceback would be great).

**Comment by Carreau** [comment]
And did not meant to close, sorry wrong button.

**Comment by Phillip-M-Feldman** [comment]
I tried putting the following into my ipython configuration file:

    ip= get_ipython()
    ip.highlighting_style= 'fruity'

When I start ipython, I get the following error:

    NameError: name 'get_ipython' is not defined

I'll be grateful for any advice.

**Comment by Carreau** [comment]
Yeah, the configuration files cannot refer to IPython instance as IPython does not exists yet. 

The way I do things it to usually grep for given option in the help:

```
$ ipython --help-all | grep highlighting_style
--TerminalInteractiveShell.highlighting_style=<Union>
--TerminalInteractiveShell.highlighting_style_overrides=<Dict>
```

Then replace the `--` (which is how to pass the optin at the CLI) with `c.`

```
$ ipython --TerminalInteractiveShell.highlighting_style='fruity'
```

Appear to work, so the corresponding config option will be

```
c.TerminalInteractiveShell.highlighting_style='fruity'
```

**Comment by Phillip-M-Feldman** [comment]
This is tremendous!  Thanks!
